Dockerfile

```diff
@@ -89,10 +89,25 @@ RUN set -x && apt-get update \
   && cd /opt/intel/crypto-api-toolkit \
   # disable building tests
   && sed -i -e 's;test;;g' ./src/Makefile.am \
+  # disable enclave signing inside CTK
+  && sed -i -e '/libp11SgxEnclave.signed.so/d' ./src/p11/trusted/Makefile.am \
   && ./autogen.sh \
   && ./configure --enable-dcap --with-token-path=/home/tcs-issuer \
   && make && make install
 
+# Sign the enclave with custom config.
+COPY enclave-config enclave-config
+ENV SGX_SIGN=/opt/intel/sgxsdk/bin/x64/sgx_sign
+RUN set -x; cd /opt/intel/crypto-api-toolkit/src/p11/trusted \
+  && ${SGX_SIGN} gendata -enclave ./.libs/libp11SgxEnclave.so.0.0.0 -out /tmp/libp11SgxEnclave.unsigned -config /opt/intel/enclave-config/p11Enclave.config.xml \
+  && /opt/intel/enclave-config/sign-enclave.sh -in /tmp/libp11SgxEnclave.unsigned -out /tmp/libp11SgxEnclave.signature -keyout /tmp/publickey.pem \
```

Comment:
What happens to the public key? Do we leave it on the container so someone can validate the signature using the same container image?

Reply:
Currently, I am not saving the public key inside the enclave, do you suggest we should do? If understood correctly, that get's saved as part of a signed enclave metada.

Reply:
I think we should save it, because someone might need it to check the enclave signature during attestation. Of course if the key comes from outside it could be done out-of-band, but having the public key as part of the container might be user-friendly, especially if the signing key is ever changed.

Reply:
I fixed the PR, so that now it saves the enclave public-key in the final image under `/usr/local/share/`.